apps/rush-lib/src/logic/operations/OperationExecutionManager.ts

```diff
@@ -548,4 +588,131 @@ export class OperationExecutionManager {
     this._terminal.writeStdoutLine(leftPart + rightPart);
     this._terminal.writeStdoutLine('');
   }
+
+  /**
+   * Print a more detailed timeline and analysis of CPU usage for the build.
+   */
+  private _printTimeline(): void {
+    //
+    // Gather the operation records we'll be displaying
+    //
+
+    const operations: OperationExecutionRecord[] = [];
+
+    for (const operation of this._executionRecords) {
+      if (operation.stopwatch.startTime && operation.stopwatch.endTime) {
+        operations.push(operation);
+      }
+    }
+
+    operations.sort((a, b) => a.stopwatch.startTime! - b.stopwatch.startTime!);
+
+    //
+    // Determine timing for all tasks (wall clock and execution times)
+    //
+
+    const allStart: number = operations[0].stopwatch.startTime!;
+    const allEnd: number = Math.max(...operations.map((operation) => operation.stopwatch.endTime!));
+    const allDuration: number = allEnd - allStart;
+    const workDuration: number = operations
+      .map((operation) => operation.stopwatch.duration)
+      .reduce((sum, value) => sum + value);
+
+    //
+    // Do some calculations to determine what size timeline chart we need.
+    //
+
+    const maxNameLength: number = Math.max(...operations.map((operation) => operation.name.length));
```

Comment:
If the number of operations in the graph gets into the thousands this will throw an exception by overflowing the stack. I'd recommend refactoring all of these calculations that involve the entire array into a single loop across the array that computes all of the values in a single pass, without throwaway intermediate arrays or dumping arrays onto the stack.